flink-runtime/src/test/java/org/apache/flink/runtime/state/v2/StateBackendTestV2Base.java

```diff
@@ -177,8 +182,8 @@ protected <K> AsyncKeyedStateBackend<K> restoreAsyncKeyedBackend(
                                 new JobID(),
                                 "test_op",
                                 keySerializer,
-                                numberOfKeyGroups,
-                                new KeyGroupRange(0, numberOfKeyGroups - 1),
+                                keyGroupRange.getNumberOfKeyGroups(),
+                                keyGroupRange,
```

Comment:
why do we need both parameteres - if one can be derived from another?

Reply:
In this test `keyGroupRange` can be derived from `numberOfKeyGroups`.
In practice, `numberOfKeyGroups` is come from `MaxNumberOfParallelSubtasks`, it may be different from `keyGroupRange.getNumberOfKeyGroups()`, so we do need both parameters.